test/unit/specs/components/download-button.spec.js

```diff
@@ -0,0 +1,84 @@
+import DownloadButton from '~/components/DownloadButton'
+import render from '../../test-utils/render'
+import { mount } from '@vue/test-utils'
+import axios from 'axios'
+
+import local from '~/utils/local'
+
+jest.mock('~/utils/local', () => ({
+  get: jest.fn(),
+  set: jest.fn(),
+}))
+
+const mockFilesizes = {
+  'MP3 98kbs': 1000,
+  'MP3 VBR': 1500,
+  FLAC: 2500,
+  OGG: 2000,
+}
+
+jest.mock('axios', () => ({
+  head: jest.fn((extensionName) => mockFilesizes[extensionName]),
+}))
+
+const formats = [
+  {
+    extension_name: 'MP3 98kbs',
+    download_url: 'https://mp3d.jamendo.com/download/track/1532771/mp31/',
+  },
+  {
+    extension_name: 'MP3 VBR',
+    download_url: 'https://mp3d.jamendo.com/download/track/1532771/mp32/',
+  },
+  {
+    extension_name: 'FLAC',
+    download_url: 'https://mp3d.jamendo.com/download/track/1532771/flac/',
+  },
+  {
+    extension_name: 'OGG',
+    download_url: 'https://mp3d.jamendo.com/download/track/1532771/ogg/',
+  },
+]
+
+const doRender = async () => {
+  const wrapper = render(DownloadButton, { propsData: { formats } }, mount)
+  await DownloadButton.fetch.call(wrapper.vm)
+  return wrapper
+}
+
+describe('DownloadButton', () => {
+  beforeEach(() => {
+    axios.head.mockReset()
+  })
+
+  // TODO(@sarayourfriend) convert this to testing-library once it's possible to get the `vm` from the wrapper
```

Comment:
This is blocked by this PR https://github.com/testing-library/vue-testing-library/pull/250